app/controllers/NewUkimsNumberController.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.TraderProfileConnector
+import controllers.actions._
+import forms.UkimsNumberFormProvider
+import models.Mode
+import navigation.Navigator
+import pages.{NewUkimsNumberPage, UkimsNumberPage}
+import play.api.data.{Form, FormError}
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import views.html.{NewUkimsNumberView, UkimsNumberView}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class NewUkimsNumberController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: UkimsNumberFormProvider,
+  traderProfileConnector: TraderProfileConnector,
+  val controllerComponents: MessagesControllerComponents,
+  view: UkimsNumberView,
+  newView: NewUkimsNumberView
+)(implicit ec: ExecutionContext)
+    extends BaseController {
+
+  private val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] =
+    (identify andThen getData andThen requireData) { implicit request =>
+      val preparedForm = request.userAnswers.get(UkimsNumberPage) match {
```

Comment:
does this need to be NewUkimsNumberPage?

Reply:
This is supposed to pre-populate the form with the existing ukims number